src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.SingleSegment.Tests.cs

```diff
@@ -0,0 +1,132 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using MicroBenchmarks.Serializers;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public partial class Utf8JsonReaderTests
+    {
+        private const int Iterations = 1_000_000;
+
+        private byte[] s_singleEmptyMultiLineComment;
+        private byte[] s_multipleEmptyMultiLineComment;
+        private byte[] s_singleShortMultiLineComment;
+        private byte[] s_multipleShortMultiLineComment;
+        private byte[] s_singleLongMultiLineComment1;
+        private byte[] s_singleLongMultiLineComment2;
+
+        private byte[] s_singleEmptySingleLineComment;
+        private byte[] s_multipleEmptySingleLineComment;
+        private byte[] s_singleShortSingleLineComment;
+        private byte[] s_multipleShortSingleLineComment;
+        private byte[] s_singleLongSingleLineComment;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            // every test case ends with {} so that we can re-use it for both Skip and Allow options
+            s_singleEmptyMultiLineComment = Encoding.UTF8.GetBytes("/**/{}");
+            s_multipleEmptyMultiLineComment = Encoding.UTF8.GetBytes("/**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**/{}");
+            s_singleShortMultiLineComment = Encoding.UTF8.GetBytes("/*asdasd*/{}");
+            s_multipleShortMultiLineComment = Encoding.UTF8.GetBytes("/*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*/{}");
+            s_singleLongMultiLineComment1 = Encoding.UTF8.GetBytes("/*" + new string('c', 2000) + "*/{}");
+            s_singleLongMultiLineComment2 = Encoding.UTF8.GetBytes("/*" + new string('\n', 2000) + "*/{}");
+
+            s_singleEmptySingleLineComment = Encoding.UTF8.GetBytes("//\n{}");
+            s_multipleEmptySingleLineComment = Encoding.UTF8.GetBytes("//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n{}");
+            s_singleShortSingleLineComment = Encoding.UTF8.GetBytes("//asdasd\n{}");
+            s_multipleShortSingleLineComment = Encoding.UTF8.GetBytes("//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n{}");
+            s_singleLongSingleLineComment = Encoding.UTF8.GetBytes("//" + new string('c', 2000) + "\n{}");
+
+            MultiSegmentSetup();
+        }
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void SingleEmptyMultiLineComment_CommentHandling_Allow() => GenericReadTest(s_singleEmptyMultiLineComment, JsonCommentHandling.Allow);
```

Comment:
The comment handling enum (allow and skip) can be Params instead, to avoid code duplication.

Maybe something like:
```C#
[Params(JsonCommentHandling.Skip, JsonCommentHandling.Allow)]
public JsonCommentHandling CommentHandling;
```